core/src/main/java/ai/timefold/solver/core/impl/solver/ConsumerSupport.java

```diff
@@ -58,6 +69,12 @@ void consumeFinalBestSolution(Solution_ finalBestSolution) {
         if (bestSolutionConsumer != null) {
             scheduleIntermediateBestSolutionConsumption();
         }
+        // Make sure the first initialized solution is consumed before the final best solution consumer shutdown the executor.
```

Comment:
@rsynek, some tests indicate that the first initialized consumer may not run because the final solution starts and shuts down the executor before its execution.

I believe that may not occur in production because the LS wouldn't finish before the first initialized solution is consumed. Let me know if you think I should update the tests instead.

https://github.com/TimefoldAI/timefold-solver/actions/runs/9668030108/job/26671216009?pr=915
https://github.com/TimefoldAI/timefold-solver/actions/runs/9668030300/job/26671219202?pr=915